### ISSUE TITLE
Backport (2.1): Exec shell and view logs do not work in an upgraded setup

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#21948]
https://github.com/rancher/rancher/pull/21924

**Comment by jiaqiluo** [comment]
backport https://github.com/rancher/rancher/issues/21919

**Comment by jiaqiluo** [comment]
cannot exec shell in v2.1.12-rc2

**Comment by sangeethah** [comment]
We have also seen the kubectl shell not work for RKE clusters after upgrade.

**Comment by cjellick** [comment]
fix will be available in v2.1.12-rc3, but dax's fix is going int 2.1 as well, so you might want to wait for v2.1.12-rc4

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher:v2.1.12-rc4

Steps:
- run Rancher:v2.1.11 in HA mode
- add a custom cluster with 1 control plane, 3 etcd, 2 worker nodes, k8s v1.13.5
- upgrade Rancher to v2.1.12-rc4
- deploy a workload in the default project
- after the workload is active, click to  execute shell 
- also exec the shell of an old existing workload

Results:
- for both workloads, the shell can be connected successfully 
- run the kubectl and see the following
```
> kubectl get clusterroles | grep proxy
proxy-clusterrole-kubeapiserver                                        26m
```
